packages/block-library/src/navigation-link/index.php

```diff
@@ -152,12 +152,6 @@ function render_block_core_navigation_link( $attributes, $content, $block ) {
 	$has_submenu = count( $block->inner_blocks ) > 0;
 	$is_active   = ! empty( $attributes['id'] ) && ( get_the_ID() === $attributes['id'] );
 
-	$class_name = ! empty( $attributes['className'] ) ? implode( ' ', (array) $attributes['className'] ) : false;
-
-	if ( false !== $class_name ) {
-		$css_classes .= ' ' . $class_name;
-	}
-
```

Comment:
I was seeing some of the class names being duplicated, which is why I removed this. I can't see where they're being added originally though.

With `$attributes['className']`:

![image](https://user-images.githubusercontent.com/1645628/139655342-fc5d8726-d52b-44ec-b223-a7678d406420.png)

Without `$attributes['className']`:

![image](https://user-images.githubusercontent.com/1645628/139655419-f46fa8d7-4a24-4b36-a8ba-431f2a9a1362.png)

`faq-class` (a custom test class I've added from Appearance > Menus), `menu-item`, `menu-item-type-post_type`, and `menu-item-object-page` are all duplicated when `$attributes['className']` is used here. If I change the attribute name to something else (in both `navigation-link/index.php` and `navigation/index.php`), then the class names are only included once, so I'm assuming something else is picking up on `className` and adding them. If I var_dump `$attributes['className']`, it only includes unique classes, and I'm struggling to find where the others are added.

Ideally, I'd like to leave `navigation-link/index.php` unchanged and fix the duplication problem. If you have time, it would be great if you have any ideas around this!

Reply:
Ah, yep, you're right, good catch. It seems like a bug in trunk that a custom class name is applied twice (happens when using the navigation block normally in the block editor too), so this would be a fix.

> I can't see where they're being added originally though.

I think it must be via the `get_block_wrapper_attributes` function, which is defined in WordPress core:
https://github.com/WordPress/wordpress-develop/blob/master/src/wp-includes/class-wp-block-supports.php#L179

Since the block supports both generated and custom class names, those values get merged in like this:
https://github.com/WordPress/wordpress-develop/blob/b45c85a405ab0b1968380a79f7a76dc9293adc0f/src/wp-includes/block-supports/custom-classname.php#L44-L56

In that case it looks good to me, will approve the PR

Reply:
Ahh brilliant, I understand, thanks so much for explaining. Nice, unexpected bug fix!

Thanks for your help with this! 🎉